src/transformers/processing_utils.py

```diff
@@ -85,6 +99,60 @@ def __init__(self, *args, **kwargs):
 
             setattr(self, attribute_name, arg)
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes this instance to a Python dictionary.
+
+        Returns:
+            `Dict[str, Any]`: Dictionary of all the attributes that make up this processor instance.
+        """
+        output = copy.deepcopy(self.__dict__)
+
+        # Get the kwargs in `__init__`.
+        sig = inspect.signature(self.__init__)
+        # Only save the attributes that are presented in the kwargs of `__init__`.
+        output = {k: v for k, v in output.items() if k in sig.parameters and k not in self.__class__.attributes}
+
+        output["processor_type"] = self.__class__.__name__
```

Comment:
We have `image_processor_type` and `feature_extractor_type` being used, but `processor_class` is used in [src/transformers/models/auto/processing_auto.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/auto/processing_auto.py)

Note that `processor_class` is checked if being in the [image processor/feature extraoctor] file or if in the model config file, not the new [processor] file (yet).

@amyeroberts I think it's best to keep `processor_class` in the new [processor] file instead of `processor_type`. Do you have any opinion here?